[review] MIP-60 : Bridge designs (#60)
movementlabsxyz/MIP: MIP/MIP-60/README.md

[MIP-58](https://github.com/movementlabsxyz/MIP/pull/58/files) is an instance of a _lock/mint_ bridge. Once initiated on the L1 (source chain), the user does not need to interact with the L2 (target chain) to complete the transfer. The bridge operator is responsible for bundling/coupling the lock/mint transactions.  (Similarly for the opposite direction, the user only needs to interact with the L2 chain and the bridge operator is responsible for the coupling of the burn/unlock transactions.)

## Security

### Security - HTLC-based design

See [MIP-39](https://github.com/movementlabsxyz/MIP/blob/mip-move-bridge-architecture/MIP/mip-39/README.md) for a complete discussion of the security of this type of design.

### Security - Trusted-Relayer design

See [Trusted Relayer design](#trusted-relayer-design).

> Assuming we have a _correct_ implementation of the lock/mint contracts on the source and target chains, and a trusted relayer, we may still have to deal with the following issues:

Author: These requirements are too specific to the design solution and they should be transferred to MIP-58..

> # MIP-60: Crosschain bridge architectures

Reviewer: Since this is mainly definitional, I would suggest framing this as an MIP which introduces MGs.

- **Description**: This MIP provides some background on crosschain bridges designs and trade-offs related to security, decentralization, and performance.
- **Authors**: [Franck Cassez](mailto:franck.cassez@movementlabs.xyz), [Andreas Penzkofer](mailto:andreas.penzkofer@movementlabs.xyz)

## Abstract

Crosschain bridges allow assets to be transferred between different blockchains. This MIP provides some background on crosschain bridge designs and the trade-offs related to security, decentralization, and performance.

## Motivation

> To transfer assets from Ethereum to Move Rollup (and back), we have to employ a _bridge_. Ideally we would re-use an existing bridge, but there is no available bridge template from Ethereum to Move-based Rollups, so we have to design one.

Author: ```suggestion
To transfer assets from Ethereum to the chain (and back), we have to employ a _bridge_. Ideally we would re-use an existing bridge, but there is no available bridge template from Ethereum to Move-based Rollups, so we have to design one.
```

Designing a bridge between two blockchains is a complex task. As pointed out in [Bridges – Ethereum Foundation](https://ethereum.org/en/developers/docs/bridges/#integrating-bridges):

> 1. **Building your own bridge** – Building a secure and reliable bridge is not easy, especially if you take a more trust-minimized route. Moreover, it requires years of experience and technical expertise related to scalability and interoperability studies. Additionally, it would require a hands-on team to maintain a bridge and attract sufficient liquidity to make it feasible.

To design our bridge, it may be useful to understand the main types of architectures, their strengths and weaknesses. It can also help us reflect on the different discussions and proposals (MIPs, MDs) and how they relate to the different architectures.

## Definitions and Bridge introduction

Bridge terminology is frequently ambiguous as the following shows but we attempt to clarify the terms used in this MIP.

> **Base classes of bridges** [[5]][REF5]

Reviewer: IMO it would be good to have the dimensions/attributes along which these vary listed in a table s.t. we can see how we arrive at this particular number of classes.

Author: i dont think this is a short task thus it is out of scope. The bullet points and the reference should give an entry point for further reading. If there is lateron a challenge of the name, such as if we switch to a validator-based bridge, this also can be a starting point here.

There are several base classes of bridges to apply to our setting. We select **Native Bridge** as the most appropriate for our use case.

> - **Native Bridges** bootstrap liquidity of a base chain into a different chain, e.g. to onboard users and move value into a L2 network.

Author: ```suggestion
- **Native Bridges** bootstrap liquidity from a base chain into a different chain, e.g. to onboard users and move value into a chain.
```

- **Canonical Bridges** is a term typically used to refer to L1 to L2 bridges, and have a similar meaning to Native Bridges. They typically take a longer period of time to bridge assets than liquidity networks as they require a time period for finality to be established, such as the challenge period for optimistic
rollups. 

> - **Validating Bridges** are bridges between Ethereum and a rollup, where the bridge contract verifies the state updates proposed by the offchain system.

Author: ```suggestion
- **Validating Bridges** are bridges between Ethereum and a chain, where the bridge contract verifies the state updates proposed by the off-chain system.
```

> [!NOTE]
> The term **Native Bridge** is used to refer to a bridge that transfers the native ERC-20 `$L1MOVE` token from a base chain (L1) to the `$L2MOVE` token on a different chain (L2).

**Liquidity Networks** [[5]][REF5]

The term of Liquidity Networks is associated with 

- **Continuous Liquidity Pools** The majority of Liquidity Networks consist of two liquidity pools of assets - one on the source and the other on the destination chain. In this structure, an intermediary token can act as an exchange medium allowing accurate amounts of the final token to be transferred to the user and to rebalance the pools once a transaction has occurred, e.g. RUNE in Thorchain, or stablecoins such as USDC in Stargate. 

In our setting the intermediary IS the final token, and this could be interpreted as a **lock/unlock** bridge.

- **Continuous Liquidity Pools (CLP) with AMMs** are a subset of Continuous Liquidity Pools that use Automated Market Makers (AMMs) in series. [MIP-40](https://github.com/movementlabsxyz/MIP/pull/40) has proposed such a design.

**Types of bridge mechanisms**

> There are four types of crosschain bridges. They differ in the way they handle the native assets (source chain) and the representation of the assets (wrapped asset) on the target chain.

Author: ```suggestion
There are three types of crosschain bridges. They differ in the way they handle the native assets (source chain) and the representation of the assets (wrapped asset) on the target chain.
```

1. **burn/mint**: the asset is burned on the source chain and an asset is minted on the target chain.
_Supply: The bridge operator is in control of creating supply on both chains._
1. **lock/mint**: the native asset (minted e.g. in an ERC20 contract) is locked on the source chain and a wrapped asset is minted on the target chain (which is L2 in our case).
_Supply: The bridge operator is only in control of minting supply on the target chain (L2). On L1 the bridge operator holds a token pool into/from which tokens get locked/unlocked._

> 1. **lock/unlock**: although difficult to classify in literature, this variant naturally completes the above two types. It may be classified as a subcategory of a swap, a **liquidity network** (see above), or it could be interpreted as a trusted [Automated Market Maker (AMM) with a constant sum invariant](https://medium.com/@0xmirai77/curve-v1-the-stableswap-invariant-f87ad7641aa0). The asset is locked on the source chain and unlocked on the target chain. 

Reviewer: I would say any **mint** with with a cap on how much can be minted is effectively an **unlock.**

Author: effectively yes. yet it distinguishes technicaly since the lock/mint utilizes a burn mechanism, while the lock/unlock utilizes a token collection pool from which tokens are released. I added, however, a note that raises your point

### Designs

**HTLC-based design**

Our initial bridge design, [RFC-40](https://github.com/movementlabsxyz/rfcs/blob/main/0040-atomic-bridge/rfc-0040-atomic-bridge.md), is based on an _atomic swap_ (4. above).  The current architecture of our bridge, [MIP-39](https://github.com/movementlabsxyz/MIP/blob/mip-move-bridge-architecture/MIP/mip-39/README.md), borrows ideas from _lock/mint_ and _swap_ ([HTLC](https://en.bitcoin.it/wiki/Hash_Time_Locked_Contracts)):
 
- if a user wants to bridge from Ethereum to Movement Network, they have to submit two independent transactions: one on Ethereum (lock) and one on Movement Network (mint). This is not user friendly and may be a barrier to adoption.
<!-- A bridge is slightly different to a swap in the sense that the two transactions (lock and mint) are not meant to be decoupled.-->
- another issue is that the first transfer of MOVE tokens requires a user to have some _gas tokens_ (the token used to pay execution fees on the Movement Network). If the gas token is the `$L2MOVE` token, we have to implement a mechanism to sponsor the first transfer of MOVE tokens.

> **Trusted-Relayer design**

Author: ```suggestion
**Lock/Mint design**
```

> [!TIP]

> > A simple _lock/mint_, _burn/mint_ or _lock/unlock_ mechanism requires only one transaction from the user and is probably more user friendly.

Author: ```suggestion
> A simple _lock/mint_, _burn/mint_ or _lock/unlock_ mechanism requires only one transaction from the user and is probably more user friendly than the HTLC-based design.
```

A simple design that is adopted by many L2 chains is the _lock/mint_ design. A user initiates a transfer on the source chain and the bridge operator is responsible for completing the transfer on the target chain. In this design, the trust assumptions are that the contracts on the source chain and the target chain are _correct_ and implement the lock and mint operations correctly (respectively burn and unlock for the opposite direction).

> [MIP-58](https://github.com/movementlabsxyz/MIP/pull/58/files) is an instance of a _lock/mint_ bridge. Once initiated on the L1 (source chain), the user does not need to interact with the L2 (target chain) to complete the transfer. The bridge operator is responsible for bundling/coupling the lock/mint transactions.  (Similarly for the opposite direction, the user only needs to interact with the L2 chain and the bridge operator is responsible for the coupling of the burn/unlock transactions.)

Author: ```suggestion
[MIP-58](https://github.com/movementlabsxyz/MIP/pull/58/files) is an instance of a _lock/mint_ bridge. Once initiated on the source chain, the user does not need to interact with the target chain to complete the transfer. The bridge operator is responsible for coupling the lock/mint transactions for L1->L2 transfers. Similarly for the opposite direction, the bridge operator is responsible for the coupling of the burn/unlock transactions.
```

## Security

### Security - HTLC-based design

See [MIP-39](https://github.com/movementlabsxyz/MIP/blob/mip-move-bridge-architecture/MIP/mip-39/README.md) for a complete discussion of the security of this type of design.

### Security - Trusted-Relayer design

> See [Trusted Relayer design](#trusted-relayer-design).

Reviewer 2: ```suggestion
See [Trusted Relayer design](#58).
```
